psi4/src/psi4/dlpno/triples.cc

```diff
@@ -0,0 +1,1317 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2024 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "dlpno.h"
+#include "sparse.h"
+
+#include "psi4/lib3index/3index.h"
+#include "psi4/libdiis/diismanager.h"
+#include "psi4/libfock/cubature.h"
+#include "psi4/libfock/points.h"
+#include "psi4/libmints/basisset.h"
+#include "psi4/libmints/integral.h"
+#include "psi4/libmints/local.h"
+#include "psi4/libmints/matrix.h"
+#include "psi4/libmints/mintshelper.h"
+#include "psi4/libmints/molecule.h"
+#include "psi4/libmints/orthog.h"
+#include "psi4/libmints/twobody.h"
+#include "psi4/libmints/vector.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
+#include "psi4/libpsi4util/process.h"
+#include "psi4/libqt/qt.h"
+
+#include <ctime>
+#include <algorithm>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+namespace psi {
+namespace dlpno {
+
+DLPNOCCSD_T::DLPNOCCSD_T(SharedWavefunction ref_wfn, Options &options) : DLPNOCCSD(ref_wfn, options) {}
+DLPNOCCSD_T::~DLPNOCCSD_T() {}
+
+void DLPNOCCSD_T::print_header() {
+    bool t0_only = options_.get_bool("T0_APPROXIMATION");
+    std::string triples_algorithm = (t0_only) ? "SEMICANONICAL (T0)" : "ITERATIVE (T)";
+    std::string storage = (!t0_only && options_.get_bool("WRITE_TRIPLES")) ? "DISK" : "IN CORE";
```

Comment:
disk vs. in-core algorithm doesn't depend on available memory?

Reply:
This is now moved later in the code